topology/src/main/java/io/camunda/zeebe/topology/TopologyChangeAppliers.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology;
+
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.topology.state.MemberState;
+import io.camunda.zeebe.topology.state.TopologyChangeOperation;
+import io.camunda.zeebe.util.Either;
+import java.util.function.UnaryOperator;
+
+/** */
+public interface TopologyChangeAppliers {
+
+  /**
+   * @return the operation applier for the given operation
+   */
+  OperationApplier getApplier(TopologyChangeOperation operation);
+
+  interface OperationApplier {
+
+    /**
+     * Validate and initialize the operation. Returns an Either with an exception if the operation
+     * is not valid, or a function that updates the state of the member in the cluster topology
+     *
+     * @return an either
+     */
+    Either<Exception, UnaryOperator<MemberState>> init();
```

Comment:
:question: 
1. What is `init` allowed to do? The docs say it "initializes the operation" but what does this mean? Is it just about updating the member state, e.g. going from `ACTIVE` to `LEAVING`?
3. What happens if the `UnaryOperator<MemberState>` throws because it tries to make an illegal change? I guess we try and avoid that during validation?

Reply:
1. Yes. Mostly updating the member state. 
2. Assumption is that it will always succeed. It should never fail normally. If it fails, it should be a bug. In that case, it is not clear how to proceed. Retrying would not help.

Reply:
I have updated the documentation.